### PR TITLE
 * Add logging for other IAntiforgery public methods

### LINKED ISSUE #44: Add logging to antiforgery
We should add logging

Some ideas:
- Validating a token (verbose)
- Validation failure / missing token (warning)

Thinking warning is right for failures, because generally this is a app configuration issue.

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Fixes #44 
cc @dougbu and @rynowak

### CODE REVIEW

File: src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgery.cs [hunk 1]
```diff
@@ -301,6 +295,15 @@ private AntiforgeryContext GetCookieTokens(HttpContext httpContext)
...
 
+            if (antiforgeryContext.HaveGeneratedNewCookieToken)
+            {
+                _logger.NewCookieToken();
+            }
+            else
+            {
+                _logger.ReusedCookieToken();
+            }
+
```
Comment (dougbu) [inline src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgery.cs]: Remove this addition.

Would potentially logging multiple times per request. Should instead restore the block where `SaveCookieTokenAndHeader()` is called in `GetAndStoreTokens()`. Then duplicate that logic in `SetCookieTokenAndHeader()`. That way, logging can reliably be done once -- when `HaveStoredNewCookieToken` is `false`.

Note as well that this method is called in cases (`GetTokens()`) where the cookie isn't stored at all.

**Comment by dougbu** [comment]
:watch:

**Comment by ryanbrandenburg** [comment]
:up: :date: @dougbu I believe that's what you meant?

**Comment by dougbu** [comment]
@ryanbrandenburg yup, this is what I meant.

**Comment by dougbu** [comment]
:shipit:

**Pull request merged**